src/connectors/FortmaticConnector.ts

```diff
@@ -1,20 +1,76 @@
-import { ChainId } from '@dcl/schemas/dist/dapps/chain-id'
-import { ProviderType } from '@dcl/schemas/dist/dapps/provider-type'
-import { FortmaticConnector as BaseFortmaticConnector } from '@web3-react/fortmatic-connector'
+import { ConnectorUpdate } from '@web3-react/types'
+import { AbstractConnector } from '@web3-react/abstract-connector'
+import { ChainId, ProviderType } from '@dcl/schemas'
 import { getConfiguration } from '../configuration'
 
-export class FortmaticConnector extends BaseFortmaticConnector {
-  apiKeys: Record<number, string>
+export class FortmaticConnector extends AbstractConnector {
+  private readonly apiKey: string
+  private readonly chainId: number
+  private readonly rpcUrl: string
+  private fortmatic: any
 
   constructor(chainId: ChainId) {
-    const { apiKeys } = getConfiguration()[ProviderType.FORTMATIC]
+    const fortmaticConfiguration = getConfiguration()[ProviderType.FORTMATIC]
+    const supportedChainIds = Object.keys(
+      fortmaticConfiguration.apiKeys
+    ).map(key => Number(key))
+    if (!supportedChainIds.includes(chainId)) {
+      throw new Error(`Invariant error: Unsupported chainId ${chainId}`)
+    }
+    super({ supportedChainIds })
 
-    super({ chainId, apiKey: apiKeys[chainId] })
-    this.apiKeys = apiKeys
+    this.apiKey = fortmaticConfiguration.apiKeys[chainId]
+    this.chainId = chainId
+    this.rpcUrl = fortmaticConfiguration.urls[chainId]
   }
 
-  public async getApiKey(): Promise<string> {
-    const chainId = await this.getChainId()
-    return this.apiKeys[chainId]
+  public async activate(): Promise<ConnectorUpdate> {
+    if (!this.fortmatic) {
+      // @ts-ignore
+      const { default: Fortmatic } = await import('fortmatic')
+      this.fortmatic = new Fortmatic(this.apiKey, {
+        rpcUrl: this.rpcUrl,
+        chainId: this.chainId
+      })
+    }
+
+    const account = await this.fortmatic
+      .getProvider()
+      .enable()
+      .then((accounts: string[]): string => accounts[0])
+
+    return {
+      provider: this.fortmatic.getProvider(),
+      chainId: this.chainId,
+      account
+    }
+  }
+
+  public getApiKey(): string {
+    return this.apiKey
+  }
+
+  public async getProvider(): Promise<any> {
+    return this.fortmatic.getProvider()
+  }
+
+  public async getChainId(): Promise<number | string> {
+    return this.chainId
+  }
+
+  public async getAccount(): Promise<null | string> {
+    return this.fortmatic
+      .getProvider()
+      .send('eth_accounts')
+      .then((accounts: string[]): string => accounts[0])
+  }
+
+  public deactivate() {
```

Comment:
https://github.com/Uniswap/web3-react/blob/v6/packages/fortmatic-connector/src/index.ts#L63

`deactivate` is empty on the original implementation. Any reason why it calls `this.close()` here?